lib/storage/metadata/file/MetadataFileServer.js

```diff
@@ -187,9 +198,24 @@ class MetadataFileServer {
             },
             batch: (request, logger, callback) => {
                 const dbPath = request.db.split(SUBLEVEL_SEP);
-                const subDb = dbService.lookupSubLevel(dbPath);
-                subDb.batch(request.array, SYNC_OPTIONS,
-                            err => callback(err));
+                const ops = request.array.map(
+                    op => Object.assign({}, op, { prefix: dbPath }));
+                if (this.recordLog.enabled) {
+                    this.recordLogService
+                        .withRequestParams(
+                            { logName: this.recordLog.recordLogName })
+                        .createLogRecordOps(
+                            ops, (err, logEntries) => {
+                                debug('ops to batch:', ops);
```

Comment:
To complete this answer, the `openLog` object from the env is opened on the server from the `logName` attribute sent by the client through `withRequestParams()`.